src/challenge-status/challenge-status.service.ts

```diff
@@ -85,9 +85,14 @@ export class ChallengeStatusService extends BaseService<ChallengeStatus, Challen
    * @returns {ChallengeStatus} the challenge status
    */
   async markAsFailed(gameId: string, challengeId: string, playerId: string, date: Date): Promise<ChallengeStatus> {
+    const challenge: Challenge = await this.challengeService.findById(challengeId);
     const temp: ChallengeStatus = await this.findByChallengeIdAndPlayerId(challengeId, playerId);
     const result: ChallengeStatus = await this.patch(temp.id, { state: StateEnum.FAILED, endedAt: date });
 
+    if (challenge.mode == Mode.TIME_BOMB && result.endedAt.getTime() > date.getTime()) {
```

Comment:
What is this for?

Reply:
okay I think it was meant to be in markAsRejected instead of markAsFailed. If the submission if not successful do not update the endedAt